dev/robot/install.yml

```
@@ -0,0 +1,123 @@
+---
+- name: Boot to rescue
+  hosts: all
+  gather_facts: false
+
+  vars:
+    # SSH keys to add to the server. Must already exist in Robot.
+    authorized_keys:
+      - fc:3c:b4:42:c9:bf:f9:6e:be:58:d3:12:40:c7:13:b2 # julian.toelle
+      - 21:ea:a3:5c:02:a6:0c:42:0b:ef:b5:60:ae:5b:07:9b # jonas.lammler
+
+  module_defaults:
+    group/community.hrobot.robot:
+      hetzner_user: "{{ lookup('ansible.builtin.env', 'ROBOT_USER') }}"
+      hetzner_password: "{{ lookup('ansible.builtin.env', 'ROBOT_PASSWORD') }}"
+
+  tasks:
+    - name: Upload CI SSH Key # noqa args[module]
+      delegate_to: localhost
+      community.hrobot.ssh_key:
+        name: hcloud-cloud-controller-manager
+        public_key: ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIA/+QBJNjrtqJjdsOqOwfELUS+3M3g0QRbn7pQaXfw7f ci@2024-06-27
+        state: present
+      register: ci_ssh_key
+
```

```suggestion

    - name: Set server name # noqa args[module]
      community.hrobot.server:
        server_number: "{{ server_number }}"
        server_name: "{{ inventory_hostname }}"

```
We fixed it manually, but I think we should have a task to rename the robot server to the `inventory_hostname`:

---

I would prefer to avoid meddling with the server name in Robot. Feels dirty. What do you think about inverting this and using the actual server name as the hostname instead?

```patch
diff --git a/dev/robot/install.yml b/dev/robot/install.yml
index ec191c44..598dcbd7 100644
--- a/dev/robot/install.yml
+++ b/dev/robot/install.yml
@@ -47,10 +47,15 @@
   hosts: all
   gather_facts: true
   tasks:
+    - name: Get hostname
+      community.hrobot.server_info:
+        server_number: "{{ server_number }}"
+      register: server_result
+
     - name: Deploy installimage autosetup
       ansible.builtin.copy:
         content: |
-          HOSTNAME {{ inventory_hostname }}
+          HOSTNAME {{ server_result.servers[0].server_name }}
 
           DRIVE1 /dev/sda
           DRIVE2 /dev/sdb
```

---

This would more be the role of an inventory plugin. Let's keep it as it is, and see in the future if we need it.